public/js/annotations.js

```diff
@@ -0,0 +1,255 @@
+function Annotate(params) {
+  if (!(this instanceof Annotate)) {
+    // the constructor was called without "new".
+    return new Annotate(params);;
+  }
+  params = typeof params !== 'undefined' ? params : {};
+
+  this.tool      = null;
+  this.context   = null;
+  this.callbacks = null;
+
+  // I'd like this to be styled, but this is enough for MVP
+  this.lineWidth          = params.lineWidth           || 5;
+  this.lineColor          = params.lineColor           || "#df4b26";
+  this.fillColor          = params.fillColor           || "#F49638";
+  this.highlightRadius    = params.highlightRadius     || 20;
+  this.highlightPeriod    = params.highlightPeriod     || 500;
+  this.highlightFillColor = params.highlightFillColor  || 'rgba(245, 213, 213, 0.5)';
+  this.highlightLineColor = params.highlightLineColor  || '#cc0000';
+  this.zoom               = params.zoom                || 1; // Can you just die already?
+
+  this.setActiveCanvas = function(canvas) {
+    // dereference so we can use raw DOM objects or jQuery collections
+    canvas = canvas[0] || canvas
+    if (canvas.nodeName.toLowerCase() !== 'canvas' ) {
+      throw new TypeError('Expected a DOM canvas element');
+    }
+    this.context = canvas.getContext("2d");
+
+    this.context.strokeStyle = this.lineColor;
+    this.context.fillStyle   = this.fillColor;
+    this.context.lineWidth   = this.lineWidth;
+    this.context.lineJoin    = "round";
+  }
+
+  this.erase = function() {
+    if ( this.callbacks && this.callbacks['erase'] ) {
+      try {
+        this.callbacks['erase']();
+      } catch (e) {
+        console.log('Erase callback failed. ' + e);
+      }
+    }
+
+    this.context.clearRect(0, 0, this.context.canvas.width, this.context.canvas.height);
+  }
+
+  this.draw = function(x, y) {
+    // undo the effects of the zoom
+    x = x / this.zoom;
+    y = y / this.zoom;
+
+    if ( this.callbacks && this.callbacks['draw'] ) {
+      try {
+        this.callbacks['draw'](x, y);
+      } catch (e) {
+        console.log('Draw callback failed. ' + e);
+      }
+    }
+
+    this.context.strokeStyle = this.lineColor;
+    this.context.lineTo(x, y);
+    this.context.stroke();
+  }
+
+  this.click = function(x, y) {
+    // undo the effects of the zoom
+    x = x / this.zoom;
+    y = y / this.zoom;
+
+    if ( this.callbacks && this.callbacks['click'] ) {
+      try {
+        this.callbacks['click'](x, y);
+      } catch (e) {
+        console.log('Click callback failed. ' + e);
+      }
+    }
+
+    this.context.fillStyle   = this.fillColor;
+    this.context.beginPath();
+    this.context.moveTo(x, y);
+
+    switch(this.tool) {
+      case 'leftArrow':
+        var left  = new Path2D(`m${x},${y} 40,-40 0,20 50,0 0,40 -50,0 0,20 -40,-40 z`);
+        this.context.fill(left);
+        break;
+
+      case 'rightArrow':
+        var right = new Path2D(`m${x},${y} -40,-40 0,20 -50,0 0,40 50,0 0,20 40,-40 z`);
```

Comment:
Same here. (But these two strings are the only *breaking* changes on IE. The annotations don't work, but it doesn't otherwise impact functionality.)